sound/soc/sof/ipc4-alh.h

```diff
@@ -0,0 +1,50 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ *
+ * Author: Bard Liao <yung-chuan.liao@linux.intel.com>
+ */
+
+#ifndef __SOF_INTEL_IPC4_ALH_H
+#define __SOF_INTEL_IPC4_ALH_H
+
+#define ALH_MAX_NUMBER_OF_GTW   1
+
+struct sof_gtw_attributes {
+	// Gateway data requested in low power memory.
+	uint32_t lp_buffer_alloc : 1;
+
+	//Gateway data requested in register file memory.
+	uint32_t alloc_from_reg_file : 1;
+
+	// Reserved field
+	uint32_t _rsvd : 30;
+} __packed;
+
+struct AlhMultiGtwCfg {
+	//! Number of single channels (valid items in mapping array).
+	uint32_t count;
+	//! Single to multi aggregation mapping item.
+	struct {
+		//! Vindex of a single ALH channel aggregated.
+		uint32_t alh_id;
+		//! Channel mask
+		uint32_t channel_mask;
+	} mapping[1]; //!< Mapping items
+} __packed;
+
+struct sof_alh_configuration_blob {
+	struct sof_gtw_attributes gw_attr;
+	struct AlhMultiGtwCfg alh_cfg;
+};
+
+
+#define SNDW_MULTI_GTW_BASE  0x50
```

Comment:
a comment on what this is wouldn't hurt. Not sure if this is a sizeof() or a register offset.